crates/sui/src/client_commands.rs

```diff
@@ -468,14 +468,17 @@ impl SuiClientCommands {
                 let sender = context.try_get_object_owner(&gas).await?;
                 let sender = sender.unwrap_or(context.active_address()?);
 
-                let build_config =
-                    resolve_lock_file_path(build_config, Some(package_path.clone()))?;
+                let config = resolve_lock_file_path(build_config, Some(package_path.clone()))?;
+                let run_bytecode_verifier = true;
+                let print_diags_to_stderr = true;
+
+                let sui_build_config = BuildConfig {
+                    config,
+                    run_bytecode_verifier,
+                    print_diags_to_stderr,
+                };
 
-                let resolution_graph = build_config
-                    .resolution_graph_for_package(&package_path, &mut std::io::stderr())
-                    .map_err(|err| SuiError::ModuleBuildFailure {
-                        error: format!("{:?}", err),
-                    })?;
+                let resolution_graph = sui_build_config.resolution_graph(&package_path)?;
```

Comment:
using the helper now makes this code resembles sui `BuildConfig::build(...)` more closely, i.e., driving things via `sui_framework_build::BuildConfig`)